services/gitdiff/gitdiff.go

```diff
@@ -1497,6 +1510,42 @@ func GetDiff(gitRepo *git.Repository, opts *DiffOptions, files ...string) (*Diff
 	return diff, nil
 }
 
+// GetUserSpecificDiff is like GetDiff, except that user specific data such as which files the given user has already viewed on the given PR will also be set
+func GetUserSpecificDiff(userID int64, pull *models.PullRequest, gitRepo *git.Repository, opts *DiffOptions, files ...string) (*Diff, error) {
+	diff, err := GetDiff(gitRepo, opts, files...)
+	if err != nil {
+		return diff, err
```

Comment:
```suggestion
		return nil, err
```

Reply:
I mean, it makes no difference: if `GetDiff` returns an error, the returned diff will already be `nil`. But yeah, can of course apply it.